test/simulator/timestamp/src/timestamp_manager.cpp

```diff
@@ -90,20 +92,38 @@ timestamp_manager::trim(std::string str)
 }
 
 /* Parse config string to a config map. */
-void
-timestamp_manager::parse_config(
-  const std::string &config, std::map<std::string, std::string> &config_map)
+int
+timestamp_manager::parse_config(const std::string &config,
+  std::map<std::string, std::string> &config_map, const std::vector<std::string> &supported_ops,
+  const std::vector<std::string> &unsupported_ops)
 {
     std::istringstream conf(config);
     std::string token;
 
+    if (config.empty())
+        return (0);
+
     while (std::getline(conf, token, ',')) {
         int pos = token.find('=');
         if (pos == -1)
             config_map.insert({trim(token), ""});
         else
             config_map.insert({trim(token.substr(0, pos)), trim(token.substr(pos + 1))});
     }
+
+    /* Get rid of the unsupported ops. */
+    for (std::string op : unsupported_ops) {
+        auto pos = config_map.find(op);
+        if (pos != config_map.end())
+            config_map.erase(pos);
+    }
+
+    /* Ensure that the elements in the config_map also exist in the supported_ops vector. */
+    for (const auto &op : config_map)
+        if (std::find(supported_ops.begin(), supported_ops.end(), op.first) == supported_ops.end())
+            WT_SIM_RET(EINVAL);
```

Comment:
Should we print the invalid config?

Reply:
The caller function prints the config in case of a failure.